cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/DcLineSegmentConversion.java

```diff
@@ -30,32 +34,146 @@ public DcLineSegmentConversion(PropertyBag l, Context context) {
         cconverter2 = context.dc().cgmesConverterFor(iconverter2);
     }
 
+    private static double getPAc(PropertyBag p) {
+        return Double.isNaN(p.asDouble(TARGET_PPCC)) ? 0 : p.asDouble(TARGET_PPCC); // targetPpcc is the real power injection target in the AC grid in CGMES
+    }
+
+    private void updateLossFactor1(double pAC1, double poleLossP1, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER)) {
+            iconverter1.setLossFactor((float) (poleLossP1 / pAC1) * 100);
+        } else if (pAC1 + poleLossP1 != 0) {
+            iconverter1.setLossFactor((float) (poleLossP1 / (pAC1 + poleLossP1)) * 100);
+        }
+    }
+
+    private void updateLossFactor2(double pAC2, double poleLossP2, HvdcLine.ConvertersMode mode) {
+        if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER)) {
+            iconverter2.setLossFactor((float) (poleLossP2 / Math.abs(pAC2)) * 100);
+        } else if (Math.abs(pAC2) + poleLossP2 != 0) {
+            iconverter2.setLossFactor((float) (poleLossP2 / Math.abs(pAC2) + poleLossP2) * 100);
+        }
+    }
+
+    private void updatePowerFactor(double pAC1, double pAC2) {
+        if (iconverter1.getHvdcType() == HvdcConverterStation.HvdcType.LCC) {
+            ((LccConverterStation) iconverter1).setPowerFactor(pAC1 == 0 || Double.isNaN(iconverter1.getTerminal().getQ()) ? 0.8f : (float) Math.abs(iconverter1.getTerminal().getQ() / pAC1));
+            ((LccConverterStation) iconverter2).setPowerFactor(pAC2 == 0 || Double.isNaN(iconverter2.getTerminal().getQ()) ? 0.8f : (float) Math.abs(iconverter2.getTerminal().getQ() / pAC2));
+        }
+    }
+
+    private void updateConverterStations(double pAC1, double pAC2, double poleLossP1, double poleLossP2, HvdcLine.ConvertersMode mode) {
```

Comment:
For the two else-if condition, maybe we have to add that pAC1/2 should be NaN (or 0). For me, there is a sign issue but I need our reviews. I think that when you are on inverter side, P is negative. Am I right ? And I think we have to add comments. 
````
        } else if (pAC1 != 0 && pAC2 == 0) {
            updateLossFactor1(pAC1, poleLossP1, mode);
            if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && (pAC1 - poleLossP1) != 0) {
                // pAC1 is positive.
                // lossFactor2 = poleLossP2 / pDC * 100
                // pDC = pAC1 - poleLossP1
                iconverter2.setLossFactor((float) (poleLossP2 / (pAC1 - poleLossP1)) * 100);
            } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && (Math.abs(pAC1) + poleLossP1 + poleLossP1) != 0) {
                // pAC1 is negative.
                // lossFactor2 = poleLossP2 / (pDC + poleLossP2) * 100
                // pDC = pAC1 + poleLossP1
                iconverter2.setLossFactor((float) (poleLossP2 / (Math.abs(pAC1) + poleLossP1 + poleLossP1)) * 100);
            }
        } else if (pAC2 != 0 && pAC1 == 0) {
            updateLossFactor2(pAC2, poleLossP2, mode);
            if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && (Math.abs(pAC2) + poleLossP2 + poleLossP1) != 0) {
                // pAC2 is negative.
                // lossFactor1 = poleLossP1 / pAC1 * 100
                // pAC1 = pDC + poleLossP1 = pAC2 + poleLossP2 + poleLossP1
                iconverter1.setLossFactor((float) (poleLossP1 / (Math.abs(pAC2) + poleLossP2 + poleLossP1)) * 100);
            } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && (pAC2 - poleLossP2) != 0) {
                // pAC2 is positive.
                // lossFactor1 = poleLossP1 / pDC * 100
                // pDC = pAC2 - poleLossP2
                iconverter1.setLossFactor((float) (poleLossP1 / (pAC2 - poleLossP2)) * 100);
            }
````

Reply:
> For the two else-if condition, maybe we have to add that pAC1/2 should be NaN (or 0).

Sonar counts it as bad practice ("code smells") if we explicit an implied condition e.g.

```java
if (a && b) {
// do something
} else if (a) {
// first condition is false so we necessarily have a && !b, else we are in case 1
} else if (b) {
// we necessarily have !a && b, else we are in case 1
}
```
Should we willingly do a bad practice for readability ? I'm open to all answers, I'm just not sure you were aware of that.

> For me, there is a sign issue but I need our reviews. I think that when you are on inverter side, P is negative. Am I right ? And I think we have to add comments.
> 
> ```
>         } else if (pAC1 != 0 && pAC2 == 0) {
>             updateLossFactor1(pAC1, poleLossP1, mode);
>             if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && (pAC1 - poleLossP1) != 0) {
>                 // pAC1 is positive.
>                 // lossFactor2 = poleLossP2 / pDC * 100
>                 // pDC = pAC1 - poleLossP1
>                 iconverter2.setLossFactor((float) (poleLossP2 / (pAC1 - poleLossP1)) * 100);
>             } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && (Math.abs(pAC1) + poleLossP1 + poleLossP1) != 0) {
>                 // pAC1 is negative.
>                 // lossFactor2 = poleLossP2 / (pDC + poleLossP2) * 100
>                 // pDC = pAC1 + poleLossP1
>                 iconverter2.setLossFactor((float) (poleLossP2 / (Math.abs(pAC1) + poleLossP1 + poleLossP1)) * 100);
>             }
>         } else if (pAC2 != 0 && pAC1 == 0) {
>             updateLossFactor2(pAC2, poleLossP2, mode);
>             if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && (Math.abs(pAC2) + poleLossP2 + poleLossP1) != 0) {
>                 // pAC2 is negative.
>                 // lossFactor1 = poleLossP1 / pAC1 * 100
>                 // pAC1 = pDC + poleLossP1 = pAC2 + poleLossP2 + poleLossP1
>                 iconverter1.setLossFactor((float) (poleLossP1 / (Math.abs(pAC2) + poleLossP2 + poleLossP1)) * 100);
>             } else if (mode.equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && (pAC2 - poleLossP2) != 0) {
>                 // pAC2 is positive.
>                 // lossFactor1 = poleLossP1 / pDC * 100
>                 // pDC = pAC2 - poleLossP2
>                 iconverter1.setLossFactor((float) (poleLossP1 / (pAC2 - poleLossP2)) * 100);
>             }
> ```

For sign convention, see the remarks above.

Reply:
Okay, no bad practice. I understand that NaN is not used, 0 is prefered. 
Do you treat the case PAC1 = 0 and PAC2 = 0 ?